### PR TITLE
Add solid-query and skeleton for contributors

### PR DESCRIPTION
**Pull request by Joe2k**
Solid query is awesome. Ive used react query in previous projects and its too good. Helps with inbuilt caching as well. And helps to write better code and comes with out of box data, loading and error handling.

So integrating it here to use it further for tournaments where there will be multiple same read calls.

So this will help to cache and reduce hits on our server

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -35,7 +35,9 @@
     "webpack-cli": "^5.1.4",
     "webpack-dev-server": "^4.15.1"
   },
-  "dependencies": {},
+  "dependencies": {
```
Comment (Joe2k) [inline frontend/package.json:38]: @punchagan Just noticed this.. Should I install as a dev dependency here?

Comment (punchagan) [reply frontend/package.json:38]: For a project that's being completely built as a bundle, I wasn't sure what the difference between dev dependencies and actual dependencies would be. If we really care about it, we could separate out things at some point.

**Comment by punchagan** [comment]
Ooh, I had briefly looked at the `Suspense` stuff, but didn't know about `solid-query`. This looks great. I like the data fetching, loading and error handling stuff! 

I didn't read up much, but what do you mean by in-built caching? Is it between different page navigations, after the site has been loaded once? Does it also cache between page reloads?

File: frontend/package.json (hunk 1 above)
Comment (Joe2k) [reply frontend/package.json:38]: ahh cool even im not well versed at that. Will keep like this for now i guess

**Comment by Joe2k** [comment]
https://www.bigbinary.com/blog/react-query

check this out..

There is refetch for window focus and all which i used before which will be too good for tournament updates on the go when you dont even need to refresh page

**Comment by punchagan** [comment]
> There is refetch for window focus and all which i used before which will be too good for tournament updates on the go when you dont even need to refresh page

Nice this is great! You can also set an auto refresh interval. :)

I pushed a tiny change to turn off `refreshOnWindowFocus` for the contributors query. We can live with a little stale data there. :)

**Pull request merged**